### ISSUE TITLE
Service process to speed up repeated compilations - F#

### ISSUE THREAD

**Issue by Jand42** [dotnet-websharper/core#1131]
First, on the F# side

* Parametrize compileMain to take a factory function for F# checker from the outside.
* New project wsfscservice.exe will run next to wsfsc.exe but can serve as a persistent background process, that takes compile arguments through a pipe, runs compileMain itself, returns standard/error outputs and exit code (depends on #1130).
* Build script: add a step before Clean to kill any running wsfscservice from build folder before wiping the build folder.
* Optional: add new wsconfig.json bool parameter: incrementalBuild. Default true uses wsfscservice, setting it to false runs compileMain directly.

**Comment by mrtank** [comment]
WIG and sitelets+html+bundle operations should not happen in wsfsc.exe, which need current assembly to be loaded to execute and then that version of the dll is locked in

**Comment by mrtank** [comment]
"Optional: add new wsconfig.json bool parameter: incrementalBuild. Default true uses wsfscservice, setting it to false runs compileMain directly."

This become `--standalone` flag. Default false uses wsfscservice. Also controllable `<WebSharperStandalone>True<WebSharperStandalone>` project property, or `WebSharperBuildService` environment variable = "False". All of them disables using the wsfscservice.

**Comment by Jand42** [comment]
As part of this, compiler service needs to use an AssemblyLoadContext whenever instantiating types from compiled project or it's dependencies.
These usages are:
* WS compilation: macros, generators
* Unpack website resources: only for DownloadResources e292d9a2b628e7ec1a5a5efd54d890f5c1efecee
* Html offline site: main functionality for instantiating IWebSite/Sitelet object e292d9a2b628e7ec1a5a5efd54d890f5c1efecee
* Bundling, including webworker bundles: for resource instances 1cf8aae85bebcd4195c2f2e5fdd3c54d8314b457
* WIG: main functionality, for instantiating Assembly definition

**Comment by Jand42** [comment]
Service is implemented for WebSharper 5.0-preview builds, create new ticket for the `ws` helper #1166
